### PR TITLE
Fix upgrade with complex objects. Support pagination, sorting.

### PR DESCRIPTION
**Pull request by gbevan**
* Fixes updates with complex objects (e.g. github profile in a sessions collection). (I don't really like this fix - you have a comment saying need to find better approach with arangojs iirc - I agree - I may revisit this at some point...)
* Added support for $select field selection.
* Added support for pagination
* Added debugging using debug module (this really helped)
* Added support for sorting

### CODE REVIEW

File: lib/adapter.js [hunk 1]
```diff
@@ -174,24 +184,70 @@ module.exports = (function() {
...
+      debug('find() query:', query);
+
+      db.query(query, function(e, r){
         if (e){
+          debug('find() query err:', e);
           return cb(e);
         }
-        var processor = new Processor(connections[connectionName].collections);
-        return cb(null, processor.cast(collectionName, r));
+//        debug('find() result:', r);
```
Comment (gabriel-letarte) [inline lib/adapter.js]: Remove or re-activate

File: lib/adapter.js [hunk 2]
```diff
@@ -174,24 +184,70 @@ module.exports = (function() {
...
+            let nv = {};
+            ['_id', '_key', '_rev'].forEach((k) => { nv[k] = v[k] });
+
+            searchCriteria.select.forEach((sk) => {
+              nv[sk] = v[sk];
+            });
+            return nv;
+          })
+          .then((newr) => {
+//            debug('find() select mapped results:', newr);
```
Comment (gabriel-letarte) [inline lib/adapter.js]: ditto

File: lib/adapter.js [hunk 3]
```diff
@@ -301,14 +364,16 @@ module.exports = (function() {
...
 
       var db = connections[connectionName].db;
       var col = db.collection(collectionName);
 
       var qfor = 'FOR x IN ' + collectionName;
       var qfilter = this._queryFormatter(searchCriteria.where, '==', 'FILTER');
       var qupdate = 'UPDATE { _key: x._key, ' +
-        this._queryFormatter(values, ':') + ' } IN ' + collectionName;
+        this._updateStringify(values) + ' } IN ' + collectionName;
+//        this._queryFormatter(values, ':') + ' } IN ' + collectionName;
```
Comment (gabriel-letarte) [inline lib/adapter.js]: remove?

**Comment by gabriel-letarte** [comment]
Small comments. Looks good. Would you mind squashing?

**Comment by gbevan** [comment]
cleaned up.  sure go ahead and squash the commits on merge.

**Pull request merged**